src/rdkafka.h

```diff
@@ -3330,6 +3330,56 @@ int rd_kafka_queue_poll_callback (rd_kafka_queue_t *rkqu, int timeout_ms);
 
 
 /**
+ * @name Admin interface
+ *
+ */
+
+/**
+ * @brief Delete topic(s) in cluster.
+ *
+ * @param topics List of topic (names) to delete. Per-topic errors are
+ *        returned in the \c err field of each topic.
+ * @param timeout_ms Time for both controller lookup and actual topic deletion
+ *                   on the broker. If < 0 the absolute value is used
+ *                   for controller lookup timeout only and no timeout is
+ *                   used on the broker - the topic will only be marked for
+ *                   deletion. If > 0 the remaining time after controller
+ *                   lookup will be used as DeleteTopics request timeout
+ *                   on the broker, depicting how long the broker will
+ *                   wait for all replicas to have deleted the topic
+ *                   before responding to the client.
+ *                   Despite timeout setting, the topic will be marked
+ *                   for deletion.
+ *
+ * @returns RD_KAFKA_RESP_ERR_NO_ERROR on success,
+ *          RD_KAFKA_RESP_ERR__TRANSPORT if no controller was found in
```

Comment:
RD_KAFKA_RESP_ERR_REQUEST_TIMED_OUT  doesn't describe the request failed, topic, in most case (if not all), should effectively have be marked as deletion and deleted when possible